include/ck/utility/data_type.hpp

```diff
@@ -996,6 +1411,18 @@ inline __host__ __device__ constexpr bhalf_t type_convert<bhalf_t, float>(float
     return uint16_t(u.int32 >> 16);
 }
 
+template <>
+inline __host__ __device__ constexpr int8_t type_convert<int8_t, int4_t>(int4_t x)
+{
+    return int8_t(x);
+}
+
+template <>
+inline __host__ __device__ constexpr int4_t type_convert<int4_t, int8_t>(int8_t x)
+{
+    return int4_t(x & 0x7);
+}
+
 template <typename T>
 struct NumericLimits
```

Comment:
We need implement NumericLimits for `int4_t`  since for Reduction operation Min/Max/AMax,  we are supposed to directly compare `int4_t` values in LDS/Vgpr, no need to convert them to `int32_t`

With the same reason, also `abs` need be implemented for `int4_t` in `utility/math_v2.hpp`

Reply:
Done